### ISSUE TITLE
Thoughts about same namespace but different provider?

### ISSUE THREAD

**Issue by pawamoy** [basherpm/basher#40]
What about the following situation?

```
basher install github.com/namespace/project
basher install gitlab.com/namespace/project
```

The whole point of using namespace to uniquely identify packages is defeated by the possibility to use different sources (github.com, gitlab.com, bitbucket.com, etc.).

Can something be done regarding this :confused:?

**Comment by juanibiapina** [comment]
I quickly though about it but dismissed it as very unlikely. It's possible to implement this but it seems such an edge case that I didn't think it was worth it.

Do you have any practical problems?

**Comment by pawamoy** [comment]
Not really, I just happen to have switched my repositories official location to GitLab, and my namespace there is different.

So when my libraries source themselves with Basher's `include` function (or else), I have to choose the namespace once and for all, tiding my libraries to the particular Git provider holding that namespace. 

Installing from a mirror location (like GitHub) holding a different namespace for the same libraries would then break the source chain. This would of course easily be fixed by the end user by renaming the installed directory though, or creating a symlink to it.

**Comment by juanibiapina** [comment]
But even though the same project is hosted in two places, there's no easy way to tell they're the same project if they don't have the exact `namespace/project` combination. In this case it's just saner to treat them as different projects.

So I don't have a solution for you, except maybe put the namespace in a variable?

**Comment by pawamoy** [comment]
Right, I forgot I could still use conditions and variables in my code. I feel dumb now :D I'll come up with something, and add it here if it's worth it. Thanks for your thoughts, feel free to close :slightly_smiling_face:
